doc/sphinx/src/nested_par_for.rst

```diff
@@ -86,3 +87,117 @@ GPUs.
 ``#pragma omp simd`` to vectorize the loop, which typically gives better
 vectorization loops than ``PAR_LOOP_INNER_LAYOUT=TVR_INNER_LOOP`` on
 CPUs and so is the default on CPUs.
+
+
+Performance Considerations
+---------------------------
+
+Hierarchical parallelism can produce very performant code, but a
+deeper awareness of how hardware is mapped to threads is required to
+get optimal performance. Here we list a few strategies/considerations.
+
+* On CPU, with `SIMDFOR_INNER_LOOP` you may have trouble vectorizing
+  unless you help the compiler along. One way to do so is to work with
+  raw pointers to contiguous memory, rather than working with views
+  and strides. Even for stencil ops, if you can pull out pointers that
+  represent the different points on the stencil, this can help with
+  vectorization.
+* Similarly on CPUs, due to the cost of starting up a vector op,
+  vectorization will only be a performance win if there's enough work
+  in the inner loop. A minimum of 16 points is required for the op to
+  vectorize at all. Experience shows, however, that at least 64 is
+  really required to see big wins. One strategy for providing enough
+  vector cells in the inner loop is to do a 1D ``SIMDFOR`` inner loop
+  but combine the ``j`` and ``i`` indices by simply looping over the
+  contiguous memory in a rasterized plane on a block.
+* On Cuda GPUs, the outer loop typically maps to warps, while the
+  inner maps to threads. To see good performance, you must both
+  provide enough work in the inner loop to fill a warp and enough work
+  in the outer loop to fill all warps.
```

Comment:
```suggestion
* On Cuda GPUs, the outer loop typically maps to blocks, while the
  inner maps to threads. To see good performance, you must both
  provide enough work in the inner loop to fill a streaming multiprocessor (SM) with multiple warps and enough work
  in the outer loop to create enough blocks to fill all SMs on the GPU. The number of warps in flight on the inner loop per SM (which is related to "occupancy") will depend positively on length of the inner loop and negatively on higher shared memory usage (or scratch memory  in Kokkos parlance) and higher register usage.
```

I think this is more accurate to the hardware but might be too verbose. Has this been inline with your performance testing?

Reply:
Nice thanks for clarifying this. I added an additional sentence summarizing your comment on AMD GPUs as well.

Reply:
> Has this been inline with your performance testing?

Yeah you're spot on.